### ISSUE TITLE
weights from the from list connector are ignored (i think)

### ISSUE THREAD

**Issue by alan-stokes** [NeuralEnsemble/PyNN#450]
Hi All

more attempts to get a pynn 0.8 interface for the SPiNNaker platform. I was looking that the interaction between connectors and synapse types to try to see how to link them with common code for pynn 0.7 and 0.8 
one big difference is that the weights and delays have moved into the synapse type over from the connector on most of the connectors. But the fromList and fromFile connector states that it can still handle weights and delays in their parameters, but those weights and delays are completely ignored from the projections view point for generating the parameter space object from what i can see (I admit i might be not seeing the whole picture, the PyNN code is very hard to follow). 

So my question is, is this intentional? 
if it is:
     why? 
if it isnt:
    cant we not just remove the weights and delays from the From List / File connector and have a direct relationship between weights in synapse types and connectors connections connection1 = weight 1 etc. Given that the weights and delays are currently ignored (if im right), its pointless data???? and confusing to end users???

If we cant remove that, can we have a explicit ruling over who's weights are the ones to use, for example the code below:

   weight1 = 1
   weight2 = 6
   delay1 = 2
   delay2 = 5
    pop_conn_list = [(n, (n + 2) % num_neurons, weight1, delay1) for n in range(num_neurons)]
    sim.Projection(pop, pop, sim.FromListConnector(pop_conn_list),
                   sim.StaticSynapse(weight=weight2, delay=delay2),
                   receptor_type="excitatory")

which weight and delay is to be used? the connectors of weight = 1, delay 2 or the staticSynapse with weight 6, delay 5?

Now if i'm missing something critical/obvious, please tell me, as I'm trying to wire this all together and so far I feel more lost than happy.

Alan

**Comment by apdavison** [comment]
Hi Alan,

Weights and delays in `FromListConnector` and `FromFileConnector` over-ride values from the `SynapseType` object. For example:
```python

import pyNN.neuron as sim

sim.setup()

p = sim.Population(3, sim.IF_cond_exp())

connector = sim.FromListConnector([(0, 0, 0.123, 0.1),
                                   (1, 1, 0.234, 0.2),
                                   (1, 2, 0.345, 0.3),
                                   (2, 0, 0.456, 0.4)])
                                  # by default column_names = ["weight", "delay"]

connector2 = sim.FromListConnector([(0, 0, 0.123),
                                    (1, 1, 0.234),
                                    (1, 2, 0.345),
                                    (2, 0, 0.456)],
                                   column_names=["weight"])

synapse = sim.StaticSynapse(weight=0.999, delay=1.5)

prj = sim.Projection(p, p, connector, synapse)
prj2 = sim.Projection(p, p, connector2, synapse)

print("Weights and delays from Connector:")
print(prj.get(["weight", "delay"], format="list"))
print("Weights from Connector, delays from StaticSynapse:")
print(prj2.get(["weight", "delay"], format="list"))
```
prints
```
Weights and delays from Connector:
[(0, 0, 0.123, 0.1), (2, 0, 0.456, 0.4), (1, 1, 0.234, 0.2), (1, 2, 0.345, 0.3)]
Weights from Connector, delays from StaticSynapse:
[(0, 0, 0.123, 1.5), (2, 0, 0.456, 1.5), (1, 1, 0.234, 1.5), (1, 2, 0.345, 1.5)]
```

The implementation of this is here:
https://github.com/NeuralEnsemble/PyNN/blob/master/pyNN/connectors.py#L530
```python
        for tgt, l, r in zip(local_targets, left, right):
            sources = self.conn_list[l:r, 0].astype(numpy.int)
            connection_parameters = deepcopy(projection.synapse_type.parameter_space)
            connection_parameters.shape = (r - l,)
            for col, name in enumerate(self.column_names, 2):
                connection_parameters.update(**{name: self.conn_list[l:r, col]})
```
we copy `synapse_type.parameter_space` then update it from `conn_list`.

**Comment by alan-stokes** [comment]
Hi Andrew, 

thank you for that, means I can work a bit more happy. 

regards
Alan
